security/pkg/nodeagent/sds/sdsservice.go

```diff
@@ -184,23 +184,23 @@ func (s *sdsservice) StreamSecrets(stream sds.SecretDiscoveryService_StreamSecre
 			// request's <token, resourceName, Version>, then this request is a confirmation request.
 			// nodeagent stops sending response to envoy in this case.
 			if discReq.VersionInfo != "" && s.st.SecretExist(con.conID, resourceName, token, discReq.VersionInfo) {
-				log.Debugf("Received SDS ACK from %q, connectionID %q, resourceName %q, versionInfo %q\n", discReq.Node.Id, con.conID, resourceName, discReq.VersionInfo)
+				log.Infof("Received SDS ACK from %q, connectionID %q, resourceName %q, versionInfo %q\n", discReq.Node.Id, con.conID, resourceName, discReq.VersionInfo)
 				continue
 			}
 
 			if firstRequestFlag {
-				log.Debugf("Received first SDS request from %q, connectionID %q, resourceName %q, versionInfo %q\n",
+				log.Infof("Received first SDS request from %q, connectionID %q, resourceName %q, versionInfo %q\n",
 					discReq.Node.Id, con.conID, resourceName, discReq.VersionInfo)
 			} else {
-				log.Debugf("Received SDS request from %q, connectionID %q, resourceName %q, versionInfo %q\n",
+				log.Infof("Received SDS request from %q, connectionID %q, resourceName %q, versionInfo %q\n",
 					discReq.Node.Id, con.conID, resourceName, discReq.VersionInfo)
 			}
 
 			// In ingress gateway agent mode, if the first SDS request is received but kubernetes secret is not ready,
 			// wait for secret before sending SDS response. If a kubernetes secret was deleted by operator, wait
 			// for a new kubernetes secret before sending SDS response.
 			if s.st.ShouldWaitForIngressGatewaySecret(con.conID, resourceName, token) {
```

Comment:
why are we still calling this? as I previously mentioned this is what was previously causing node-agent getting stuck. can we put it under some for of feature flag at least when the fallback secret is defined?

Reply:
ShouldWaitForIngressGatewaySecret will check whether a fallback secret exists, if a fallback secret exists, then it returns false.
If customer does not create a fallback secret, then it returns true indicating to hold the request until a real gateway secret is created. This is for backward compatibility.